src/ykman.ts

```diff
@@ -0,0 +1,2354 @@
+const completionSpec: Fig.Spec = {
+  name: "ykman",
+  description: "Configure your YubiKey via the command line",
+
+  subcommands: [
+    {
+      name: "info",
+      description:
+        "Displays information about the attached YubiKey such as serial number, firmware version, capabilities, etc",
+      options: [
+        {
+          name: ["-c", "--check-fips"],
+          description:
+            "Check if YubiKey is in FIPS Approved mode (available on YubiKey 4 FIPS only)",
+        },
+        {
+          name: ["-h", "--help"],
+          description: "Show info usage information",
+        },
+      ],
+    },
+    {
+      name: "list",
+      description: "List connected YubiKeys",
+      options: [
+        {
+          name: ["-s", "--serials"],
+          description:
+            "Output only serial numbers, one per line (devices without serial will be omitted)",
+        },
+        {
+          name: ["-r", "--readers"],
+          description: "List available smart card readers",
+        },
+        {
+          name: ["-h", "--help"],
+          description: "Show list usage information",
+        },
+      ],
+    },
+    {
+      name: "config",
+      description: "Enable or disable applications",
+      subcommands: [
+        {
+          name: "mode",
+          description: "Manage connection modes (USB Interfaces)",
+          args: {
+            name: "MODE",
+            description:
+              "MODE can be a string, such as 'OTP+FIDO+CCID', or a shortened form: 'o+f+c'. It can also be a mode number",
+          },
```

Comment:
Would love to add suggestions to this. Because it takes predefined values but they're separated by `+` we need to use a generator:
```suggestion
          args: {
            name: "MODE",
            description:
              "MODE can be a string, such as 'OTP+FIDO+CCID', or a shortened form: 'o+f+c'. It can also be a mode number",
            generators: {
              getQueryTerm: "+",
              custom: () => Promise.resolve(["OTP", "FIDO", "CCID"]),
            }
          },
```

Reply:
I get this error message when copying this code:
``` 
Type 'Promise<string[]>' is not assignable to type 'Promise<Suggestion[]>'.  Type 'string[]' is not assignable to type 'Suggestion[]'.    Type 'string' has no properties in common with type 'Suggestion'. typescript (2322) [55, 29]
```
No idea why, could you help me?

Reply:
Ah, sorry - totally forgot that in a generator it has to be a `Suggestion` object. You can make it `{ name: "OTP" }`, etc.

Reply:
I'm sorry, but I didn't understand at all. What has to be a `Suggestion` object? I'm trying to find another spec that use some generator to see if I can get it.

Reply:
Ah, my bad, should've been clearer.

The `Suggestion` type is what everything inherits from. It has all the `name`, `description`, `displayName`, etc... properties. So instead of returning strings, you can return Suggestions.

I don't think there's a particular reason that suggestions _couldn't_ return strings, will add this to my to-dos for the month :)

Reply:
I've commited some code, could you see if it's right? My mind is telling that is definitely wrong...

Do I have to create a generator like [this](https://github.com/withfig/autocomplete/blob/master/src/gpg.ts#L1), for example?

Reply:
@vaaleyard here we want a generator that triggers on each + but also that removes previously inserted tokens each time the suggestions are shown. Currently we do not provide any sugar to do this so the generator has to be written manually.

```
const generator: {
  trigger: "+",
  custom: async (tokens, executeShellCommand) => {
    const previouslyAdded = tokens[tokens.length - 1].split("+")
    // this has to be an alias map of the possible suggestions
    const possibleSuggestionsMap = new Map([
      ["OTP", ["o"]],
      ["o", ["OTP"]],
      // ...all other values
    ])
    const suggestions = new Set(possibleSuggestionsMap.values())
    
    for (const token of previouslyAdded) {
      suggestions.delete(token)
      const aliases = possibleSuggestionsMap.get(token)
      if (aliases) {
        aliases.forEach(v => { suggestions.delete(v) })
      }
    }
    return [...suggestions].map(v => ({ name: v }))
  },
}
```

Also consider this: ![image](https://user-images.githubusercontent.com/38510000/183262244-799c279b-762f-4529-b33a-978801a05f69.jpeg)

I will add a new property as a sugar for this problem this week!

Reply:
I guess instead of
```js
      if (aliases) {
        aliases.forEach(v => { suggestions.delete(v) })
      }
```

should be
```js
      if (aliases) {
        suggestions.delete(aliases)
      }
```
?

because it only works with []string.

Reply:
That's correct @vaaleyard

Reply:
`Set.prototype.delete` accepts a single primitive as argument, not an array🤔

Reply:
@fedeci there's no more errors now 😅. Could you check if it's all ok? thanks